resolver.go

```diff
@@ -32,7 +32,7 @@ type txtEntry struct {
 }
 
 func NewResolver(url string) *Resolver {
-	if !strings.HasPrefix(url, "https://") {
+	if !(strings.HasPrefix(url, "https:") || strings.HasPrefix(url, "http:")) {
```

Comment:
If I understand [RFC 8484](https://datatracker.ietf.org/doc/html/rfc8484) correctly, DoH is only defined for HTTPS, not for HTTP. Or am I mistaken here?